sites/public/pages/account/edit.tsx

```diff
@@ -0,0 +1,349 @@
+    }
+  }
+
+  const onEmailSubmit = async (data: { email: string }) => {
+    const { email } = data
+    setEmailAlert(null)
+    try {
+      await userService.update({
+        body: { ...profile, email },
+      })
+      setEmailAlert({ type: "success", message: `${t("account.settings.alerts.emailSuccess")}` })
+    } catch (err) {
+      setEmailAlert({ type: "alert", message: `${t("account.settings.alerts.generic")}` })
+      console.warn(err)
+    }
+  }
+
+  const onPasswordSubmit = async (data: {
+    password: string
+    passwordConfirmation: string
+    currentPassword: string
+  }) => {
+    const { password, passwordConfirmation, currentPassword } = data
+    setPasswordAlert(null)
+    if (passwordConfirmation !== password) {
+      setPasswordAlert({ type: "alert", message: `${t("account.settings.alerts.passwordMatch")}` })
+      return
+    }
+    try {
+      await userService.update({
+        body: { ...profile, password, currentPassword },
+      })
+      setPasswordAlert({
+        type: "success",
+        message: `${t("account.settings.alerts.passwordSuccess")}`,
+      })
+    } catch (err) {
+      const { status } = err.response || {}
+      if (status === 401) {
+        setPasswordAlert({ type: "alert", message: `Invalid current password. Please try again.` })
+      } else {
+        setPasswordAlert({ type: "alert", message: `${t("account.settings.alerts.generic")}` })
+      }
+      console.warn(err)
+    }
+  }
+
+  return (
+    <RequireLogin signInPath="/sign-in" signInMessage={t("t.loginIsRequired")}>
+      <FormsLayout>
+        <FormCard>
+          <div className="form-card__lead text-center border-b mx-0">
+            <Icon size="2xl" symbol="settings" />
+            <h2 className="form-card__title">Account Settings</h2>
+          </div>
+          <SiteAlert type="notice" dismissable />
+          <Form id="update-name" onSubmit={handleSubmit(onNameSubmit)}>
+            {nameAlert && (
+              <AlertBox type={nameAlert.type} onClose={() => setNameAlert(null)} inverted closeable>
+                {nameAlert.message}
+              </AlertBox>
+            )}
+            <div className="form-card__group border-b">
+              <label className="field-label--caps" htmlFor="firstName">
+                {t("application.name.yourName")}
+              </label>
+
+              <Field
+                controlClassName="mt-2"
+                name="firstName"
+                placeholder={`${t("application.name.firstName")}`}
+                error={errors.firstName}
+                errorMessage={t("errors.firstNameError")}
+                register={register}
+                defaultValue={profile ? profile.firstName : null}
+              />
+
+              <Field
+                name="middleName"
+                placeholder={`${t("application.name.middleNameOptional")}`}
+                register={register}
+                defaultValue={profile ? profile.middleName : null}
```

Comment:
The `middleName ` property in the profile is also optional, so I think we need to use: `profile?.middleName`